tools/c7n_mailer/c7n_mailer/email_delivery.py

```diff
@@ -28,11 +29,22 @@ class EmailDelivery(object):
     def __init__(self, config, session, logger):
         self.config      = config
         self.logger      = logger
+        self.session     = session
         self.aws_ses     = session.client('ses', region_name=config.get('ses_region'))
         self.ldap_lookup = self.get_ldap_connection()
 
     def get_ldap_connection(self):
         if self.config.get('ldap_uri'):
+            try:
+                if self.config.get('ldap_bind_password', None) and self.config.get('ldap_bind_password_in_kms', True):
+                    kms = self.session.client('kms')
+                    self.config['ldap_bind_password'] = kms.decrypt(
+                        CiphertextBlob=base64.b64decode(self.config['ldap_bind_password']))[
+                            'Plaintext']
+            except Exception as error:
+                self.logger.warning(
```

Comment:
a couple of things, blanket exceptions are a red flag in code, its better to check for the Exception and condition we explicitly expect (ie ClientError with specific code for non encrypted text) and let the program exit otherwise (permission denied), there is no correct functioning without connection credentials. second, rather introduce additional config option here, i'd just try the decrypt.

Reply:
So instead of handling the variable `ldap_bind_password_in_kms` like it was in handle.py we should just try the decrypt always? If it fails to decrypt and throws the error `InvalidCiphertextException`, then resume processing as though the password were in plaintext. For every other error on http://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html#API_Decrypt_Errors, let it fail completely.